jlc_kicad_tools/jlc_lib/generate_bom.py

```diff
@@ -69,33 +67,32 @@ def GenerateBOM(input_filename, output_filename, opts):
         footprints.add(c.getFootprint())
 
     # Check part numbers for uniqueness
-    if len(lcsc_part_numbers) == 0 and not opts.include_all_groups:
+    if len(lcsc_part_numbers) == 0:
       if opts.warn_no_partnumber:
-        _LOGGER.logger.warning("No LCSC part number found for components {}".format(",".join(refs)))
-      continue
-    elif len(lcsc_part_numbers) > 1:
-      _LOGGER.logger.error("Components {components} from same group have different LCSC part numbers: {partnumbers}".format(
+        logging.warning("No LCSC part number found for components {}".format(",".join(refs)))
+      if not opts.all_parts_lcsc:
```

Comment:
Where is this defined?

Reply:
I'm very sorry about all my confusing changes. I beg your pardon.
I'm going to do it onec again from startup. And before desturbing you again, I will try and test all the present options.
Thank you very much for your reading,
kind reagards from Cologne, Germany
yours Johannes

Reply:
No worries. It looks like you are working from an old version. The "--include-all-component-groups" version was added somewhat recently, and if you were working from an old commit you have to rebase it to the current HEAD before sending a pull request, because otherwise your changes will also revert all the changes between your starting commit and HEAD.